src/Compilers/Core/Portable/Syntax/GreenNode.cs

```diff
@@ -185,7 +175,7 @@ internal GreenNode GetRequiredSlot(int index)
         // for slot counts >= byte.MaxValue
         protected virtual int GetSlotCount()
         {
-            return _slotCount;
+            throw ExceptionUtilities.Unreachable();
```

Comment:
nothing shoudl hit this *unless* SmallSlotCount is 15.  And that should only be for SyntaxListWithManyChildren, which is the only case that overrides this.  so this is much better to mark as unreachable (note: vb was already doing this.  this just pulls up to the base).

Reply:
It feels like the comment on this method should be updated

Reply:
And, please, capture the explanation for the behavior in a comment as well

Reply:
Sgtm